train_diffusion_prior.py

```diff
@@ -137,6 +163,17 @@ def train(image_embed_dim,
             wandb.log({"Training loss": loss.item(),
                         "Steps": step,
                         "Samples per second": samples_per_sec})
+            # Log cosineSim(text_embed,predicted_image_embed) - cosineSim(text_embed,image_embed)
+            # Use NUM_TEST_EMBEDDINGS samples from the test set each time
+            # Get embeddings from the most recently saved model
+            diff_cosine_sim = report_cosine_sims(diffusion_prior,
+                    image_reader,
+                    text_reader,
+                    train_set_size,
+                    val_set_size,
+                    NUM_TEST_EMBEDDINGS,
+                    device)
+            wandb.log({"Cosine similarity difference": diff_cosine_sim})
```

Comment:
Let's run this once every N steps (default N = 100 )
And let's do that as well for the computation of the validation loss instead of only at the end like now